src/codeu/chat/client/commandline/Chat.java

```diff
@@ -23,12 +23,29 @@
 import codeu.chat.client.core.UserContext;
 import codeu.chat.common.*;
 import codeu.chat.util.Tokenizer;
+import codeu.chat.util.Uuid;
 
 public final class Chat {
 
   private static File logFile;
   private static PrintWriter pw_log;
 
+  //used to access Chat's users from the user panel for interest system feature
+  private Context rootPanelContext;
+  //used to access Chat's conversations from outside the user panel
+  private UserContext userPanelContext;
+
+  private HashMap<Uuid, Set<Uuid>> userInterestMap = new HashMap<>();
```

Comment:
Also this change can be made later to make sure you guys get done on time, but I think it would clean things up a bit to move it there.